### PR TITLE
ADR 072: Re-instate a request-for-comments archive.

### PR DESCRIPTION
**Pull request by creachadair**
This ADR proposes to restore a variation of the old Request for Comments
documentation that we previously used. The proposal differs from the original
formulation, and does not replace ADRs.

### CODE REVIEW

File: docs/architecture/adr-072-request-for-comments.md [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+
+   Instead, an RFC will exist to document background, articulate general
+   principles, and serve as a historical record of discussion and motivation.
+
+   In this system, an RFC may _only_ result in a decision indirectly, via ADR
+   documents created in response to the RFC.
+
+Just so that there is a consistent format, I also propose that:
+
+-  RFC files are named `rfc-XXX-title.{md,txt}` and are written in plain text
```
Comment (tychoish) [inline docs/architecture/adr-072-request-for-comments.md:50]: also + `rst`

File: docs/architecture/adr-072-request-for-comments.md [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+question, and documents the consensus answer to that question.
+
+Some discussions are more open-ended, however, or don't require a specific
+decision to be made (yet). Such conversations are still valuable to document,
+and several members of the Tendermint team have been doing so by writing gists
+or Google docs to share them around. That works well enough in the moment, but
+gist and docs are hard to discover after the fact. Moreover, these kinds of
+open-ended discussions are very useful to people who come later -- either as
+new team members or as outside contributors seeking to use and understand the
+thoughts behind our architectural decisions.
```
Comment (tychoish) [inline docs/architecture/adr-072-request-for-comments.md:31]: I also like the group editing thing that isn't possible with gists.

**Review by tychoish — APPROVED** [review]

File: docs/rfc/README.md [hunk 3]
```diff
@@ -0,0 +1,5 @@
+---
+order: 1
+parent:
+  order: false
+---
```
Comment (cmwaters) [inline docs/rfc/README.md:5]: I'm guessing the table of contents will go here. Do we also want to add a few lines that reiterate what you've said in this ADR in describing what are the conditions of an RFC and the expected process

**Comment by cmwaters** [comment]
Do you also mind adding a reference to this ADR and ADR071 in https://github.com/tendermint/tendermint/tree/master/docs/architecture#table-of-contents

File: docs/architecture/adr-072-request-for-comments.md (hunk 2 above)
Comment (creachadair) [reply docs/architecture/adr-072-request-for-comments.md]: I added some language to highlight that.

File: docs/rfc/README.md (hunk 3 above)
Comment (creachadair) [reply docs/rfc/README.md:5]: Yes, I left this blank for now till we discussed it. I'll fill it in before merging.

File: docs/architecture/adr-072-request-for-comments.md (hunk 1 above)
Comment (creachadair) [reply docs/architecture/adr-072-request-for-comments.md]: Added.

File: docs/rfc/rfc-template.md [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+
+> Any context or orientation needed for a reader to understand and participate
+> in the substance of the Discussion. If necessary, this section may include
+> links to other documentation or sources rather than restating existing
+> material, but should provide enough detail that the reader can tell what they
+> need to read to be up-to-date.
+
+### References
+
+> Links to external materials needed to follow the discussion may be added here.
```
[the remark this answers is not in the document]
Comment (creachadair) [reply docs/rfc/rfc-template.md:23]: I added a note to that effect.

**Comment by williambanfield** [comment]
Thanks for putting this up. I definitely agree that a more 'ideas gathering' document would be very useful. I'm wondering two things as I read through this that are pretty related to each other:
1) Should the writer of the RFC attempt to capture the discussion in the RFC? As @tychoish pointed out, that may take a lot of extra work and might not be worth it.
2) What is the end state of such an RFC? i.e. When should an author actually merge the document into the repository?

**Comment by creachadair** [comment]
> 1. Should the writer of the RFC attempt to capture the discussion in the RFC? As @tychoish pointed out, that may take a lot of extra work and might not be worth it.

I intentionally didn't stipulate who should do this. I'm reluctant to put a lot of process rules here, since there is no decision at risk. That said, the idea is that "somebody" (the author is a good choice, or someone else affiliated with the PR where the RFC initially lands) should synthesize the discussion. 

The case @tychoish raised was I think more serious than we are likely to see in an RFC: We were talking about mailing list threads, which serialize a potentially-long and potentially quite parallel discussion. An RFC doesn't need to be a single PR and done for all time—the author can just merge it, and subsequent PRs can modify it. 

If you agree with that formulation, I would welcome suggestions for how to capture that concisely.

> 2. What is the end state of such an RFC? i.e. When should an author actually merge the document into the repository?

I'd argue the RFC itself is the "end state". The initial commit of an RFC is the proposal, some amount of discussion happens, and then either it's merged or discarded.

File: docs/rfc/README.md (hunk 3 above)
Comment (creachadair) [reply docs/rfc/README.md:5]: I have updated the README with some more detail, PTAL.

**Review by williambanfield — APPROVED** [review]
Makes total sense. This seems like an improved way to store thoughts on early-stage ideas. I think we can also give this a shot and see if it works and discard if not.

**Review by tychoish — APPROVED** [review]

**Pull request merged**